pkg/tfgen/docs.go

```diff
@@ -47,32 +47,34 @@ import (
 type argumentDocs struct {
 	// The description for this argument.
 	description string
-
-	// (Optional) The names and descriptions for each argument of this argument.
-	arguments map[string]string
-
-	// Whether this argument was derived from a nested object. Used to determine
-	// whether to append descriptions that have continued to the following line
-	isNested bool
 }
 
 // Included for testing convenience.
 func (ad argumentDocs) MarshalJSON() ([]byte, error) {
-	j, err := json.Marshal(struct {
-		Description string
-		Arguments   map[string]string
-		IsNested    bool
+	return json.Marshal(struct {
+		Description string `json:"description"`
 	}{
 		Description: ad.description,
-		Arguments:   ad.arguments,
-		IsNested:    ad.isNested,
 	})
-	if err != nil {
-		return nil, err
-	}
-	return j, nil
 }
 
+type docsPath string
```

Comment:
Interesting choice, so presumably "." is not allowed in fragments; but this is not []string because using string repr buys automatic hashing/equality/printing in Go?

Please document what the fragments of this path are. Are they TF property names like "foo_bar"? What happens for a schema like this:

```
type X {
    foo []Y
}

type Y {
   bar string
}
```

What is the path of bar?

Reply:
I'll add some comments on what to expect.

`"."` is not allowed in fragments. `docsPath` segments are TF names, as written in TF docs. The path of `bar` (assuming that `X` is the top level resource) would be `foo.bar`. TF doesn't alter path names for collections (as opposed to objects).


Part of the reason is that representing as `string` gets us immutability, hashing and comparison for free. Another part is that sometimes TF docs will document a block as:
```
### `foo`
- `bar` (Optional String) a field of `foo`
```

Other providers will document the same field as:
```
### `x.foo`
- `bar` (Optional String) a field of `foo`
```

This representation lets our "detect enclosing objects from title" logic work for free.

Reply:
The fact that it's "foo.bar" instead of "foo.$.bar" or some such is surprising, worth calling out, but it may be the right choice. The rest makes sense!